google/cloud/sql/connector/instance.py

```diff
@@ -79,33 +81,34 @@ def _from_str(cls, ip_type_str: str) -> IPTypes:
         return cls(ip_type_str.upper())
 
 
+@dataclass
 class ConnectionInfo:
+    """Contains all necessary information to connect securely to the
+    server-side Proxy running on a Cloud SQL instance."""
+
+    client_cert: str
+    server_ca_cert: str
+    private_key: bytes
     ip_addrs: Dict[str, Any]
-    context: ssl.SSLContext
     database_version: str
     expiration: datetime.datetime
+    context: ssl.SSLContext | None = None
 
-    def __init__(
-        self,
-        ephemeral_cert: str,
-        database_version: str,
-        ip_addrs: Dict[str, Any],
-        private_key: bytes,
-        server_ca_cert: str,
-        expiration: datetime.datetime,
-        enable_iam_auth: bool,
-    ) -> None:
-        self.ip_addrs = ip_addrs
-        self.database_version = database_version
-        self.context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+    def create_ssl_context(self, enable_iam_auth: bool = False) -> ssl.SSLContext:
+        """Constructs a SSL/TLS context for the given connection info."""
+        if self.context is not None:
```

Comment:
Shall we add a comment to remind readers why we're caching the context? e.g., "Cache the SSL context to ensure we don't read from disk repeatedly when configuring a secure connection."

Reply:
good call, will add one